packages/docprovider/src/yprovider.ts

```diff
@@ -83,11 +75,8 @@ export class WebSocketProvider implements IDocumentProvider {
           }
         );
       })
-      .then(() => this._ready.resolve(true))
-      .catch(reason => {
-        console.warn(reason);
-        this._ready.resolve(false);
-      });
+      .then(() => this._ready.resolve())
+      .catch(reason => console.warn(reason));
```

Comment:
I would argue we should reject the ready promise in the `catch`.

Reply:
Ok then let follow the general usage pattern for this.

Reply:
I think the shorter version of that comment is to say that it might be better if it is always safe to:
```ts
await provider.ready;
```